packages/rest/src/lib/REST.ts

```diff
@@ -0,0 +1,121 @@
+import { EventEmitter } from 'events';
+import { CDN } from './CDN';
+import { InternalRequest, RequestData, RequestManager, RequestMethod } from './RequestManager';
+import { DefaultRestOptions, RESTEvents } from './utils/constants';
+
+/**
+ * Options to be passed when creating the REST instance
+ */
+export interface RESTOptions {
+	/**
+	 * The base api path, without version
+	 * @default 'https://discord.com/api'
+	 */
+	api: string;
+	/**
+	 * The cdn path
+	 * @default 'https://cdn.discordapp.com'
+	 */
+	cdn: string;
+	/**
+	 * The extra offset to add to rate limits in milliseconds
+	 * @default 50
+	 */
+	offset: number;
+	/**
+	 * The number of retries for errors with the 500 code, or errors
+	 * that timeout
+	 * @default 3
+	 */
+	retries: number;
+	/**
+	 * The time to wait in milliseconds before a request is aborted
+	 * @default 15_000
+	 */
+	timeout: number;
+	/**
+	 * Extra information to add to the user agent
+	 * @default `Node.js ${process.version}`
+	 */
+	userAgentAppendix: string;
+	/**
+	 * The version of the API to use
+	 * @default '8'
+	 */
+	version: string;
+}
+
+export class REST extends EventEmitter {
+	public readonly cdn: CDN;
+	public readonly requestManager: RequestManager;
+
+	public constructor(options: Partial<RESTOptions> = {}) {
+		super();
+		this.cdn = new CDN(options.cdn ?? DefaultRestOptions.cdn);
```

Comment:
The default is unnecessary as `CDN`'s constructor already defaults to this.
```suggestion
		this.cdn = new CDN(options.cdn);
```

Reply:
Intended `??`
![image](https://user-images.githubusercontent.com/17960496/106145961-cee7bf80-617e-11eb-93be-64a3f102ba40.png)